### PR TITLE
Aggregate flamelevels

### PR DESCRIPTION
**Pull request by jasonjkeller**
Addresses #191 

Currently, jfr-core  sends `JfrMethodSample` to NR. Those events are queried  and then aggregated by the flamegraph service to build flamegraphs.

This PR aggregates `JfrMethodSample`  into `JfrFlameLevel`.  Now, Jfr-core will only send `JfrFlameLevel` events to NR.    This results in building flamegraphs from flamelevels, which has better  fidelity and at a fraction of the data cost. 

This PR leverages the existing aggregation code in the [flamegraph service here](https://source.datanerd.us/java-agent/flamegraph-service/tree/master/src/main/java/com/newrelic/flamegraphservice/model) to aggregate `JfrMethodSample`.  Specifically, the `FrameFlattener` and `FlamegraphMarshaller`.  

The bulk of the logic  to aggregate are in the new classes `ProfileSummarizer` and `Profiler Registry`. To wire up this new registry, the trail starts in `EventConverter` where the registry is hooked up.  Within the registry,  you have the new summarizers.

Validation of data was done using Java Mission Control as the standard and then comparing Jfr-core output to that. [Here are the results.](https://docs.google.com/document/d/1oELGnEMfSbxKFzAjD8li0U2n7-4KL670223OyVSbz2A/edit#heading=h.umxvglo1zk66)

### CODE REVIEW

File: jfr-mappers/src/main/java/com/newrelic/jfr/toevent/ThreadLockEventMapper.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 import static com.newrelic.jfr.RecordedObjectValidators.hasField;
 
-import com.newrelic.jfr.MethodSupport;
+import com.newrelic.jfr.profiler.MethodSupport;
```
Comment (kittylyst) [inline jfr-mappers/src/main/java/com/newrelic/jfr/toevent/ThreadLockEventMapper.java:12]: Given that classes like this also need to use MethodSupport, why did we move it into the profiler package?

File: jfr-mappers/src/main/java/com/newrelic/jfr/profiler/ProfileSummarizer.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+  private String getFrameName(JvmStackTraceEvent.JvmStackFrame frame) {
+    StringBuilder entryBuilder = new StringBuilder();
+    entryBuilder.append(frame.getDesc());
+    entryBuilder.append(":");
+    entryBuilder.append(frame.getLine());
+    return entryBuilder.toString();
+  }
+
+  private static JvmStackTraceEvent stackTraceToStackFrames(Map<String, String> jfrStackTrace) {
```
Comment (kittylyst) [inline jfr-mappers/src/main/java/com/newrelic/jfr/profiler/ProfileSummarizer.java:160]: Should this, and similar methods, live on this class? Or would they be better elsewhere and with less restricted visibility?

File: jfr-mappers/src/main/java/com/newrelic/jfr/profiler/StackTraceEvent.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package com.newrelic.jfr.profiler;
+
+public interface StackTraceEvent {}
```
Comment (kittylyst) [inline jfr-mappers/src/main/java/com/newrelic/jfr/profiler/StackTraceEvent.java:3]: Not quite sure I see the point of this marker interface.

**Review by kittylyst — APPROVED** [review]
A few minor nits, feel free to address or not as you see fit.

File: jfr-mappers/src/main/java/com/newrelic/jfr/toevent/ThreadLockEventMapper.java (hunk 1 above)
Comment (XiXiaPdx) [reply jfr-mappers/src/main/java/com/newrelic/jfr/toevent/ThreadLockEventMapper.java:12]: 🦅  👁️  moved

File: jfr-mappers/src/main/java/com/newrelic/jfr/profiler/StackTraceEvent.java (hunk 3 above)
Comment (XiXiaPdx) [reply jfr-mappers/src/main/java/com/newrelic/jfr/profiler/StackTraceEvent.java]: good point, removed

**Pull request merged**